oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OAuth2AuthorizationServerConfigurer.java

```diff
@@ -249,4 +274,17 @@ public void configure(B builder) {
 	private static <B extends HttpSecurityBuilder<B>> KeyManager getKeyManagerBean(B builder) {
 		return builder.getSharedObject(ApplicationContext.class).getBean(KeyManager.class);
 	}
+
+	private static <B extends HttpSecurityBuilder<B>> ProviderSettings getProviderSettings(B builder) {
+		ProviderSettings providerSettings = builder.getSharedObject(ProviderSettings.class);
+		if (providerSettings == null) {
+			providerSettings = getProviderSettingsBean(builder);
```

Comment:
`ProviderSettings` will not be a required `@Bean` since we can default the values - `getProviderSettingsBean()` will throw if it's not defined

Reply:
I like the idea.

If we don't register the filter at all, and the user navigates to `/.well-known/openid-configuration` then they get an `HTTP 302` redirecting to `/login`. I'm feeling we should instead return an `HTTP 404`. Thoughts ?

Additionally, `ProviderSettings` is mutable and could change. If the issuer is added after startup, we wouldn't be registering the endpoint. Is that fine ?

I'd love a bit of guidance on writing some tests for that.